build-scripts/rcm-guest/call_puddle.sh

```diff
@@ -2,24 +2,68 @@
 #
 # Download conf file from URL and call puddle
 # Allows keeing all conf files in git and calling directly
+# call_puddle.sh [CONF_FILE_URL] [SIGNED_KEYS]
+# SIGNED_KEYS is optional and should be in the format "<key>,<key>,<key>" with no spaces
 #
 
+# set -o xtrace
 set -e
 
-if [ "$#" -lt 1 ] ; then
-  echo "call_puddle.sh requires at least a URL to a conf file"
-  echo "call_puddle.sh conf-url [PUDDLE_OPTS]"
-  echo "All other passed paramters will be given directly to puddle"
+usage()
+{
+cat<<EOF
+  usage: ${0} --conf <conf_url> --keys <sig_keys|optional> [puddle opts]
+  Options:
+    --conf <conf_url> URL of puddle conf file to load. Required.
+    --keys <sig_keys> Optional. Comma separated signature keys for RPMs to load into puddle.
+    puddle opts: Required. Any options required by puddle will be passed to that command.
+EOF
+}
+
+CONF_FILE_URL=
+SIGNED_KEYS=
+
+while [[ "$#" -ge 1 ]]; 
+do
+    case "${1}" in
+      --conf)
+        echo "CONF"
+        CONF_FILE_URL="$2"
+        shift 2;;
+      --keys)
+        echo "KEYS"
+        SIGNED_KEYS="$2"
+        shift 2;;
+      *)
+        break
+        echo "OTHER: ${1}";;
+   esac
+done
+
+if [[ -z ${CONF_FILE_URL} ]]; then
+  echo "Must provide puddle conf URL!"
+  usage
+  exit 1
+fi
+
+if [[ "$#" -eq 0 ]]; then
+  echo "Must provide options to pass to puddle!"
+  usage
   exit 1
 fi
 
+
 CONF_FILE=$(mktemp)
-curl -o "${CONF_FILE}" "${1}"
+curl -o "${CONF_FILE}" "${CONF_FILE_URL}"
 
-shift
+if [ -n "${SIGNED_KEYS}" ]; then
+  echo "$(grep "(keys[ \t]*=[ \t]*)\|(signed[ \t]*=[ \t]*)" --invert-match "${CONF_FILE}")" > "${CONF_FILE}"
```

Comment:
Weird, unlike a lot of our stuff that's hard to test, this script was easy as I could just remove the puddle call. And it worked fine every time at least locally. But just in case, I'll add the second mktemp, just to be extra sure.